browser/ui/geolocation/geolocation_utils_mac.mm

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/geolocation/geolocation_utils.h"
+
+#import <CoreLocation/CoreLocation.h>
+
+#include "base/logging.h"
+
+namespace geolocation {
+
+bool IsSystemLocationSettingEnabled() {
+  // Service is off globally.
+  if (![CLLocationManager locationServicesEnabled]) {
+    return false;
+  }
+
+  CLLocationManager* manager = [[CLLocationManager alloc] init];
+  if (@available(macOS 11.0, *)) {
+    auto status = [manager authorizationStatus];
+    LOG(ERROR) << __func__ << " ##### " << (int)status;
+    if (status == kCLAuthorizationStatusAuthorized) {
```

Comment:
With local build, manager gives always not determined state.
Need to check this with signed build.